app/models/paypal_record.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class PaypalRecord < ApplicationRecord
+  belongs_to :parent_record, class_name: "PaypalRecord", optional: true
+  has_many :child_records, class_name: "PaypalRecord", inverse_of: :parent_record, foreign_key: :parent_record_id
+
+  RECORD_TYPES = [
```

Comment:
Should these be symbols? Seems correct in ruby-land, but they seem to come back as strings when returned fro the database

Reply:
The database has to store them as strings because SQL does not have any notion of "symbols". They are strings because in MySQL-land, there are only strings and nothing else.

I think it makes more sense to store them as symbols in Ruby, because symbols generally imply more of a "this is a constant and unchangeable value" feeling, but then of course you have to pay attention during comparison because `"foo" == :foo` evaluates to `false`. You'd have to do `"foo".to_sym == :foo` or `"foo" == :foo.to_s` to yield the expected result.

Reply:
Cool, no change needed at present then